File: backend/app/models/project.py
"""Pydantic models for project CRUD operations."""

from pydantic import BaseModel
from typing import Optional


class ProjectCreate(BaseModel):
    name: str
    description: Optional[str] = None
    status: str = "on_track"
    team_id: Optional[str] = None


class ProjectUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    status: Optional[str] = None
    team_id: Optional[str] = None


class ProjectResponse(BaseModel):
    id: str
    name: str
    description: Optional[str] = None
    status: str
    team_id: Optional[str] = None
    team_name: Optional[str] = None
    created_at: Optional[str] = None


class ProjectDetailResponse(ProjectResponse):
    meeting_count: int = 0
    total_tasks: int = 0
    completed_tasks: int = 0
    overdue_tasks: int = 0
    meetings: list[dict] = []

File: frontend/src/types/project.ts
export interface Project {
  id: string;
  name: string;
  description?: string;
  status: string;
  team_id?: string;
  team_name?: string;
  created_at: string;
}

export interface ProjectDetail extends Project {
  meeting_count: number;
  total_tasks: number;
  completed_tasks: number;
  overdue_tasks: number;
  meetings: Array<{
    id: string;
    title?: string;
    meeting_date?: string;
    status: string;
  }>;
}

export interface ProjectCreate {
  name: string;
  description?: string;
  status?: string;
  team_id?: string;
}

export interface ProjectUpdate {
  name?: string;
  description?: string;
  status?: string;
  team_id?: string;
}
